Call bash with `/usr/bin/env bash` instead of `/bin/bash`

akshatagarwl opened the pull request:
On some systems `/bin/bash` is not available hence cilium doesn't compile. This changes almost all of the occurences of `/bin/bash` with `/usr/bin/env bash`

qmonnet commented on install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml:
```diff
@@ -89,7 +89,7 @@ spec:
       containers:
 {{- if .Values.sleepAfterInit }}
-      - command: [ "/bin/bash", "-c", "--" ]
+      - command: [ "/usr/bin/env", bash", "-c", "--" ]
```
Aren't you missing double quotes on this one?

akshatagarwl replied:
Ohh my bad :facepalm:

qmonnet commented on test/k8sT/manifests/cilium-ds-clean-only.yaml:
```diff
@@ -6,7 +6,7 @@ spec:
       containers:
       - name: cilium-agent
-        command: [ "/bin/bash", "-c", "--" ]
+        command: [ "/usr/bin/env", bash", "-c", "--" ]
```
Same here.

qmonnet reviewed (changes requested):
Looks good overall, please fix the two occurrences below.

Could you also please squash your two commits together?

kkourt reviewed (approved):
LGTM, thanks!

Was this done automatically via some script? If yes, as an optional change maybe you can add the command you used in the commit message?

tklauser reviewed (changes requested):
It looks like the image lint CI job is currently failing for this PR: https://github.com/cilium/cilium/pull/16495/checks?check_run_id=2801445310

This is due to the changes in `images/`. You'll also need to update the `cilium-runtime` and `cilium-builder` images, see https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images for details.

kaworu commented on test/fuzzing/oss-fuzz-build.sh:
```diff
@@ -1,3 +1,3 @@
-#/bin/bash -eu
+#!/usr/bin/env bash -eu
```
This won't work:

```console
$ ./test/fuzzing/oss-fuzz-build.sh
/usr/bin/env: ‘bash -eu’: No such file or directory
/usr/bin/env: use -[v]S to pass options in shebang lines
```

```suggestion
#!/usr/bin/env bash

set -o errexit
set -o pipefail
set -o nounset
```

akshatagarwl replied:
Fixed. Thanks

pchaigno commented on .github/workflows/build-images-ci.yaml:
```diff
@@ -104,7 +104,7 @@ jobs:
             quay.io/${{ github.repository_owner }}/${{ matrix.name }}-ci:${{ steps.tag.outputs.tag }}-race
           build-args: |
-            BASE_IMAGE=quay.io/cilium/cilium-runtime:28b5a8658b596d12d80b0e7dad3efc2e77ec2d65@sha256:fbf8eee141101fade247dbe94cf84ca3bdcd92b96108996c50859ab7edd607d0
+            BASE_IMAGE=quay.io/cilium/cilium-runtime:6e2e3f2606e478b1c9a404070e3b21f2f5496445
```
Why were the SHA256 hashes removed?

akshatagarwl replied:
Could this be an issue with the `update-cilium-runtime-image.sh` script?

tklauser replied:
This is expected. The SHA256 hash will be updated once the image build action ran for a second time and `make update-runtime-image` is run again. We might need to clarify that bit in https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

pchaigno replied:
@tklauser So they will be updated before we merge this PR, right?

tklauser replied:
@pchaigno Yes, they will need to be updated using `make -C images update-runtime-image` before merging.

pchaigno commented on .github/workflows/build-images-ci.yaml:
```diff
@@ -153,7 +153,7 @@ jobs:
             quay.io/${{ github.repository_owner }}/${{ matrix.name }}-ci:${{ steps.tag.outputs.tag }}-race
           build-args: |
-            BASE_IMAGE=quay.io/cilium/cilium-runtime:28b5a8658b596d12d80b0e7dad3efc2e77ec2d65@sha256:fbf8eee141101fade247dbe94cf84ca3bdcd92b96108996c50859ab7edd607d0
+            BASE_IMAGE=quay.io/cilium/cilium-runtime:6e2e3f2606e478b1c9a404070e3b21f2f5496445
```
This GitHub workflow runs on `pull_request_target`. You will have to open a temporary, draft PR with that trigger changed to `pull_request` to properly test this pull request. Otherwise, it will run the master version of this workflow.

akshatagarwl replied:
@pchaigno I've opened #16543. Can you have a look and tell if that seems correct?

pchaigno replied:
Looks good. I'll trigger the tests.

pchaigno replied:
I forget that won't work with a forked repository :facepalm: I'll copy your branch and will send it on my side :-)

pchaigno replied:
Testing at https://github.com/cilium/cilium/pull/16553.

pchaigno replied:
The tests at #16553 are currently failing because of the missing SHA256 hashes. We'll have to rerun once those are updated.

akshatagarwl committed:
replace /bin/bash with /usr/bin/env bash

* On some systems /bin/bash is not available hence cilium doesn't compile.
This changes almost all of the occurences of `/bin/bash` with
`/usr/bin/env bash` except for the vendor/ directory
* Also fix the test/fuzzing/oss-fuzz-build.sh script

Signed-off-by: Akshat Agarwal <akshat@accuknox.com>

akshatagarwl committed:
update cilium-{runtime,builder}

Signed-off-by: Akshat Agarwal <akshat@accuknox.com>

aanm reviewed (changes requested):
@humancalico the test failures are legitimate:

```
Error: failed to start container "cilium-agent": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/usr/bin/env bash\": stat /usr/bin/env bash: no such file or directory": unknown              
```
